python/dgl/graphbolt/itemset.py

```diff
@@ -325,3 +328,30 @@ def __getitem__(self, idx: Union[int, slice]) -> Dict[str, Tuple]:
     def names(self) -> Tuple[str]:
         """Return the names of the items."""
         return self._names
+
+    def __repr__(self) -> str:
+        return _itemset_str(self, "ItemSetDict")
+
+
+def _itemset_str(itemset: Union[ItemSet, ItemSetDict], name) -> str:
+    final_str = f"{name}("
+    indent_len = len(final_str)
+
+    def _add_indent(_str, indent):
+        lines = _str.split("\n")
+        lines = [lines[0]] + [" " * indent + line for line in lines[1:]]
+        return "\n".join(lines)
+
+    item_str = (
+        "items="
+        + _add_indent(str(list(itemset)), indent_len + len("items="))
```

Comment:
`list(itemset)` iterates over one by one and it's a bit redundant. I think just printing `ItemSet._items`, `ItemSetDict._itemsets` and `self._names` is more concise and readable.